### PR TITLE
Week 8-13 Progress Update: Nidhi Gupta 

### PR DESCRIPTION
**Pull request by nidhi98gupta**
Added progress update.
Updated project plan.

@xpdavid @LuLechuan Please review.

### CODE REVIEW

File: Nidhi_Gupta/progress.md [hunk 1]
```diff
@@ -104,35 +104,63 @@ Retrospect on Week 7:
 Code looks clean now. I guess it would take some more time to get the main PR merged. I plan to work on this along with other issues related to comments feature.
 
 Week 8 Plan:
-- [ ] Code Review
-- [ ] Ability for instructors to add comments from view, Giver > Question > Recipient
-- [ ] Ability for instructors to add comments from view, Recipient > Question > Giver
-- [ ] Add tests for both
```
Comment (xpdavid) [inline Nidhi_Gupta/progress.md]: Do not remove the original plan :D

**Comment by nidhi98gupta** [comment]
@xpdavid Ready for review.
I shouldn't have forgotten to update project plan :disappointed:

File: Nidhi_Gupta/progress.md [hunk 2]
```diff
@@ -104,35 +104,85 @@ Retrospect on Week 7:
...
+
+Retrospect on Week 10:
+This week I worked upon separating feedback participant comments from instructor comments completely. It simplified the implementation of comment indexes. I also wrote lots of unit tests for all the new methods that I introduced in my multiple PRs. I feel more confident about writing tests at the end of this week. No major difficulties encountered. The reason for not completing this week's plan is same as the week before. Unfortunately I forgot to update my project plan last week.
+
 Week 11 Plan:
-- [ ] Finish any remaining work from week 8-10
-- [ ] Fix issues after code review
+- [x] Finish any remaining work from week 8-10
+- [x] Fix issues after code review
 - [ ] Implement backend of ability for instructors to disable comments
```
Comment (nidhi98gupta) [inline Nidhi_Gupta/progress.md:154]: I updated project plan of this and following weeks before they started. (You can verify from my commits in this PR) I hope that's okay.

**Comment by nidhi98gupta** [comment]
@xpdavid Added progress report and updated project plan of following week. Please review.

**Review by xpdavid — APPROVED** [review]
Add some suggestions for further work?

**Comment by nidhi98gupta** [comment]
@xpdavid Added.

File: Nidhi_Gupta/progress.md [hunk 3]
```diff
@@ -186,3 +186,11 @@ Retrospect
 
 Week 14 Plan:
 - [ ] Improve UI of new comments feature
+
+Suggestions for further work
+- Enable feedback participant comments for other question types.
+- Ability to add feedback participant comments with response too.
```
Comment (LuLechuan) [inline Nidhi_Gupta/progress.md]: Change the first word to be an action word.

The whole plan sounds good.

File: Nidhi_Gupta/progress.md [hunk 4]
```diff
@@ -104,35 +104,93 @@ Retrospect on Week 7:
...
+Retrospect
+- This week, I worked upon improving code quality of comments PR. Fortunately I stumbled upon a bug in comments PR before the release and managed to fix it as well as add more tests. I also worked upon extending comments feature to MSQ type questions. Upon discussion with mentors, it was decided to leave the work to enable other question types to other contributors (since the task is easy and there is already a good template for them to refer to) and I can be the reviewer. Adding feedback participant comments was a huge task for me and I am really happy to see the way it shaped up in three months.
 
 Week 14 Plan:
-- [ ] Code review and code fix
+- [ ] Improve UI of new comments feature
+
+Suggestions for further work
+- Enable feedback participant comments for other question types.
+- Ability to add feedback participant comments with response too.
```
Comment (LuLechuan) [inline Nidhi_Gupta/progress.md]: Change the first word to be an action word.

The whole plan sounds good.

**Comment by nidhi98gupta** [comment]
@LuLechuan @xpdavid Done. Please merge this.

**Pull request merged**